libr/anal/class.c

```diff
@@ -1163,4 +1163,53 @@ R_API void r_anal_class_list_vtables(RAnal *anal, const char *class_name) {
 		r_cons_printf ("  %4s vtable 0x%"PFMT64x" @ +0x%"PFMT64x" size:+0x%"PFMT64x"\n", vtable->id, vtable->addr, vtable->offset, vtable->size);
 	}
 	r_vector_free (vtables);
+}
+
+static void list_all_functions_at_vtable_offset(RAnal *anal, const char *class_name, ut64 offset){
+
+	RVTableContext vtableContext;
+	r_anal_vtable_begin(anal, &vtableContext);
+	ut8 function_ptr_size = vtableContext.word_size; 
+
+	ut64 func_address;
+	RVector *vtables = r_anal_class_vtable_get_all (anal, class_name);
+	RAnalVTable *vtable;
+
+	if(!vtables) // no vtables
+		return;
+
+	r_vector_foreach (vtables, vtable) {
+		if(vtable->size < offset + function_ptr_size) {
+			continue;
+		}
+
+		if(vtableContext.read_addr(anal, vtable->addr+offset, &func_address))
+			r_cons_printf("Function address: 0x%08"PFMT64x", in %s vtable %4s \n", func_address, class_name, vtable->id);
+	}
+	r_vector_free (vtables);
+}
+
+R_API void r_anal_class_list_vtable_offset_functions(RAnal *anal, const char *class_name, ut64 offset){
+	if(!class_name) {
+		SdbList *classes = r_anal_class_get_all (anal, true);
+		SdbListIter *iter;
+		SdbKv *kv;
+		ls_foreach (classes, iter, kv) {
+			char *class_name = sdbkv_key(kv);
```

Comment:
I think thats const. Otherwise its leaking

Reply:
I peeked the sdbkv_key() function, it doesn't allocate anything, but changed it to const